Translate 'AJAX and APIs' (#140)
reactjs/pl.reactjs.org: content/docs/faq-ajax.md

---
id: faq-ajax
title: AJAX i API
permalink: docs/faq-ajax.html
layout: docs
category: FAQ
---

### Jak mogę wykonać zapytanie AJAX? {#how-can-i-make-an-ajax-call}

> Możesz użyć dowolną bibliotekę AJAX jaką chcesz. Do najpopularniejszych wyborów należy [Axios](https://github.com/axios/axios), [jQuery AJAX](https://api.jquery.com/jQuery.ajax/) oraz wbudowane w przeglądarki [window.fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API).

Reviewer: ```suggestion
Możesz użyć dowolnej biblioteki AJAX. Do najpopularniejszych wyborów należą: [Axios](https://github.com/axios/axios), [jQuery AJAX](https://api.jquery.com/jQuery.ajax/) oraz wbudowane w przeglądarki [window.fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API).
```
Za *używać* korzystamy z dopełniacza, a nie biernika (za: https://sjp.pwn.pl/poradnia/haslo/uzywac-czego;719.html i https://sjp.pwn.pl/poradnia/haslo/biernik-czy-dopelniacz;6047.html).

> ### W którym momencie cyklu życia komponentu powinienem wykonać zapytanie AJAX? {#where-in-the-component-lifecycle-should-i-make-an-ajax-call}

Reviewer: ```suggestion
### W którym momencie cyklu życia komponentu powinno się wykonać zapytanie AJAX? {#where-in-the-component-lifecycle-should-i-make-an-ajax-call}
```

> Powinieneś uzupełniać dane z wykorzystaniem zapytań AJAX w metodzie [`componentDidMount`](/docs/react-component.html#mounting) cyklu życia komponentu. Dzięki temu będziesz w stanie użyć metody `setState`, aby zmodyfikować stan komponentu po pobraniu danych.

Reviewer: ```suggestion
Dane należy uzupełniać z wykorzystaniem zapytań AJAX w metodzie [`componentDidMount`](/docs/react-component.html#mounting). Dzięki temu możliwe będzie użycie metody `setState` do zmodyfikowania stanu komponentu po pobraniu danych.
```

### Przykład: Używanie rezultatu zapytania AJAX do ustawienia lokalnego stanu {#example-using-ajax-results-to-set-local-state}

> Niniejszy przykład pokazuje jak wykonując zapytania AJAX w metodzie `componentDidMount` można zmodyfikować stan komponentu.

Reviewer: ```suggestion
Niniejszy przykład pokazuje, jak, wykonując zapytania AJAX w metodzie `componentDidMount`, można zmodyfikować stan komponentu.
```
